src/structures/interaction/Interaction.ts

```diff
@@ -0,0 +1,49 @@
+import { APIApplicationCommandInteraction, APIGuildInteraction, InteractionType, Snowflake } from 'discord-api-types';
+import { Client } from '../../clients';
+import { deconstruct } from '../../utils/Snowflake';
+import { GuildMember } from '../guild/GuildMember';
+import { User } from '../User';
+
+function isGuildInteraction(data: any): data is APIGuildInteraction {
```

Comment:
why?

Reply:
Because it can be APIGuildInteraction or APIDMInteraction